skyportal/handlers/api/spectrum.py

```diff
@@ -547,3 +577,43 @@ def get(self, obj_id):
                 )
 
         return self.success(data=return_values)
+
+
+class SpectrumRangeHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """Docstring appears below as an f-string."""
+
+        json = self.get_json()
+
+        try:
+            standardized = SpectrumRangeQuery.load(json)
+        except ValidationError as e:
+            return self.error(f'Invalid request body: {e.normalized_messages()}')
+
+        instrument_ids = standardized['instrument_ids']
+        min_date = standardized['min_date']
+        max_date = standardized['max_date']
+
+        gids = [g.id for g in self.current_user.accessible_groups]
+
+        query = (
+            DBSession()
+            .query(Spectrum)
+            .join(GroupSpectrum)
+            .filter(GroupSpectrum.group_id.in_(gids))
+        )
+
+        if instrument_ids is not None:
+            query = query.filter(Spectrum.instrument_id.in_(instrument_ids))
+        if min_date is not None:
+            utc = Time(min_date, format='isot', scale='utc')
+            query = query.filter(Spectrum.observed_at >= utc)
```

Comment:
@scizen9  This should fix the bug you are seeing vv
```suggestion
            query = query.filter(Spectrum.observed_at >= utc.isot)
```

Reply:
Tried these changes, but still does not seem to be filtering.